test/mocha/schedule.js

```diff
@@ -44,5 +44,16 @@ describe("schedule", function () {
                 assert.fail();
             });
         });
+
+        describe("Promise.setTimeoutScheduler", function() {
+            it("should throw for non function", function() {
+                try {
+                    Promise.setTimeoutScheduler({});
+                } catch (e) {
+                    return Promise.resolve();
+                }
+                assert.fail();
```

Comment:
Rather than have code that does not run that will fail the test if it is run, seems like it would be better to write the test so it explicitly succeeds if the catch() case is called.